x-pack/plugins/ingest_manager/server/services/epm/elasticsearch/template/template.ts

```diff
@@ -359,18 +363,40 @@ const updateExistingIndex = async ({
   indexTemplate: IndexTemplate;
 }) => {
   const { settings, mappings } = indexTemplate.template;
+
+  // for now, remove from object so as not to update stream or dataset properties of the index until type and name
+  // are added in https://github.com/elastic/kibana/issues/66551.  namespace value we will continue
+  // to skip updating and assume the value in the index mapping is correct
+  delete mappings.properties.stream;
+  delete mappings.properties.dataset;
+
+  // get the dataset values from the index template to compose data stream name
+  const indexMappings = await getIndexMappings(indexName, callCluster);
+  const dataset = indexMappings[indexName].mappings.properties.dataset.properties;
+  if (!dataset.type.value || !dataset.name.value || !dataset.namespace.value)
+    throw new Error(`dataset values are missing from the index template ${indexName}`);
+  const dataStreamName = `${dataset.type.value}-${dataset.name.value}-${dataset.namespace.value}`;
+
   // try to update the mappings first
-  // for now we assume updates are compatible
   try {
     await callCluster('indices.putMapping', {
       index: indexName,
       body: mappings,
     });
+    // if update fails, rollover data stream
   } catch (err) {
-    throw new Error('incompatible mappings update');
+    try {
+      const path = `/${dataStreamName}/_rollover`;
+      await callCluster('transport.request', {
+        method: 'POST',
+        path,
+      });
+    } catch (error) {
+      throw new Error(`cannot rollover data stream ${dataStreamName}`);
```

Comment:
What do you think about adding the error message that comes from `elasticsearch` to the new error message? This can be helpful for troubleshooting.